File: XJYChartDemo/XJYChartDemo/View/LineChartCell.m
//
//  HomeFiveTableViewCell.m
//  RecordLife
//
//  Created by 谢俊逸 on 17/03/2017.
//  Copyright © 2017 谢俊逸. All rights reserved.
//

#import "LineChartCell.h"
#import <XJYChart/XChart.h>
@implementation LineChartCell

- (void)awakeFromNib {
  [super awakeFromNib];
  // Initialization code
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
  [super setSelected:selected animated:animated];

  // Configure the view for the selected state
}
- (instancetype)initWithStyle:(UITableViewCellStyle)style
              reuseIdentifier:(NSString*)reuseIdentifier {
  if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
    NSMutableArray* itemArray = [[NSMutableArray alloc] init];

    NSMutableArray* numbersArray = [NSMutableArray new];

    //点的数据
    for (int j = 0; j < 2; j++) {
      NSMutableArray* numberArray = [NSMutableArray new];

      for (int i = 0; i < 5; i++) {
        int num = [[XRandomNumerHelper shareRandomNumberHelper]
                      randomNumberSmallThan:14] *
                  [[XRandomNumerHelper shareRandomNumberHelper]
                      randomNumberSmallThan:14];
        NSNumber* number = [NSNumber numberWithInt:num];
        [numberArray addObject:number];
      }

      [numbersArray addObject:numberArray];
    }

    NSArray* colorArray = @[
      [UIColor tealColor], [UIColor brickRedColor], [UIColor babyBlueColor],
      [UIColor bananaColor], [UIColor orchidColor]
    ];

    for (int i = 0; i < 2; i++) {
      XLineChartItem* item =
          [[XLineChartItem alloc] initWithDataNumberArray:numbersArray[i]
                                                    color:colorArray[i]];
      [itemArray addObject:item];
    }

    XNormalLineChartConfiguration* configuration =
        [[XNormalLineChartConfiguration alloc] init];
    configuration.lineMode = CurveLine;
    configuration.isShowShadow = YES;

    XLineChart* lineChart =
        [[XLineChart alloc] initWithFrame:CGRectMake(0, 0, 375, 200)
                            dataItemArray:itemArray
                        dataDiscribeArray:[NSMutableArray arrayWithArray:@[
                          @"January", @"February", @"March", @"April", @"May"
                        ]]
                                topNumber:@240
                             bottomNumber:@0
                                graphMode:MutiLineGraph
                       chartConfiguration:configuration];
    [self.contentView addSubview:lineChart];
    self.selectionStyle = UITableViewCellSelectionStyleNone;
  }
  return self;
}
@end

File: XJYChartDemo/XJYChartDemo-Swift/View/LineChartCell.swift
//
//  LineChartCell.swift
//  XJYChartDemo-Swift
//
//  Created by Kelly Roach on 8/17/18.
//  Copyright © 2018 JunyiXie. All rights reserved.
//

import UIKit
import XJYChart

class LineChartCell :UITableViewCell {
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
        // Configure the view for the selected state
    }

    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)

        var itemArray: [AnyHashable] = []
        var numbersArray = [[Int32]]()
        //点的数据
        for _ in 0..<2 {
            var numberArray = [Int32]()
            for _ in 0..<5 {
                let num: Int = Int(XRandomNumerHelper.shareRandomNumber().randomNumberSmallThan(14)) * Int(XRandomNumerHelper.shareRandomNumber().randomNumberSmallThan(14))
                let number = Int32(num)
                numberArray.append(number)
            }
            numbersArray.append(numberArray)
        }
        let colorArray = [UIColor.teal(), UIColor.brickRed(), UIColor.babyBlue(), UIColor.banana(), UIColor.orchid()]
        for i in 0..<2 {
            let item = XLineChartItem(dataNumber: NSMutableArray(array: numbersArray[i]), color: colorArray[i])
            itemArray.append(item!)
        }
        let configuration = XNormalLineChartConfiguration()
        configuration.lineMode = XLineMode.CurveLine
        let lineChart = XLineChart(frame: CGRect(x: 0, y: 0, width: 375, height: 200), dataItemArray: NSMutableArray(array: itemArray), dataDiscribeArray: ["January", "February", "March", "April", "May"], topNumber: 240, bottomNumber: 0, graphMode: XLineGraphMode.MutiLineGraph, chartConfiguration: configuration)
        contentView.addSubview(lineChart!)
        selectionStyle = UITableViewCellSelectionStyle.none
    
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
